tests/functional/test_snap_dcgm.py

```diff
@@ -1,86 +1,175 @@
-        ), f"Failed to set {config} to {new_value}"
+        ), f"Failed to set {config} to {value}"
 
-        # restart the service to apply the configuration
         subprocess.run(f"sudo snap restart {service}".split(), check=True)
 
-        for attempt in Retrying(wait=wait_fixed(2), stop=stop_after_delay(10)):
-            with attempt:
-                assert 0 == subprocess.call(
-                    f"nc -z localhost {value.lstrip(':')}".split()
-                ), f"{service} is not listening on {value}"
-
-    # Check new config
-    set_config_and_check(new_value)
-
-    # Revert back
-    set_config_and_check(str(old_value))
+    @classmethod
+    @retry(wait=wait_fixed(2), stop=stop_after_delay(10))
+    def check_bind_config(cls, service: str, bind: str) -> None:
+        """Check if a service is listening on a specific bind."""
+        assert 0 == subprocess.call(
+            f"nc -z localhost {bind.lstrip(':')}".split()
+        ), f"{service} is not listening on {bind}"
+
+    @classmethod
+    def get_config(cls, config: str) -> str:
+        """Check if a configuration exists in the snap configuration.
+
+        :return: The value of the current configuration
+        """
+        result = subprocess.check_output("sudo snap get dcgm -d".split(), text=True)
+        dcgm_snap_config = json.loads(result.strip())
+        assert config in dcgm_snap_config, f"{config} is not in the snap configuration"
+        return str(dcgm_snap_config[config])
+
+    @classmethod
+    @retry(wait=wait_fixed(2), stop=stop_after_delay(10))
+    def check_metric_config(cls, metric_file: str = "") -> None:
+        """Check if the metric file is loaded in the dcgm-exporter service.
+
+        :param metric_file: The metric file to check for, if empty check if nothing is loaded
+        """
+        result = subprocess.check_output(
+            "ps -eo cmd | grep '/bin/[d]cgm-exporter'", shell=True, text=True
+        )
+
+        if metric_file:
+            assert f"-f {metric_file}" in result, f"Metric file {metric_file} is not loaded"
+        else:
+            assert "-f" not in result.split(), "Metric file is loaded, but should not be"
+
+    @pytest.mark.parametrize(
+        "service, config, new_value",
+        [
+            ("dcgm.dcgm-exporter", "dcgm-exporter-address", ":9466"),
+            ("dcgm.nv-hostengine", "nv-hostengine-port", "5566"),
+        ],
+    )
+    def test_dcgm_bind_config(self, service: str, config: str, new_value: str) -> None:
+        """Test snap bind configuration."""
+        old_value = self.get_config(config)
+
+        # Valid config
+        self.set_config(service, config, new_value)
+        self.check_bind_config(service, new_value)
+
+        # Invalid config
+        self.set_config(service, config, "test")
+        _check_service_failed(f"snap.{service}")
+
+        # Revert back
+        self.set_config(service, config, old_value)
+        self.check_bind_config(service, old_value)
```

Comment:
non-blocking: isn't this redundant? We are already proving we can change the config in the lines above. Whether the parameter is old or new shouldn't matter.